### PR TITLE
First pass at porting SDL_rtf to SDL3

### PR DESCRIPTION
**Pull request by madebr**
Everything builds without warning, but I don't really know how to test `showrtf` because I don't have reference rtf files + fonts 

- remove autotools build scripts
- update cmake script to same level as other satellite libraries

Issues remaining:
- does it actually work?
- the `wikiheaders.pl` script does not generate any wiki pages + man pages. It shows no warnings.

I created a SDL2 git branch, so this pr can be merged into main.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -241,7 +240,7 @@ if(SDL2RTF_INSTALL)
 endif()
 
 if(SDL2RTF_SAMPLES)
-    add_executable(showrtf showrtf.c)
+    add_executable(showrtf samples/showrtf.c)
```
Comment (sezero) [inline CMakeLists.txt]: `samples` -> `examples` maybe? (like other satellite libs.)

Comment (madebr) [reply CMakeLists.txt]: I changed this.

Comment (sezero) [reply CMakeLists.txt]: Off-topic: in SDL_mixer, playmus and playwave sources can be similarly moved under an `examples` directory, and `SDL_mixer.sym` can be moved under `src` directory, for better consistency.

**Comment by madebr** [comment]
I just tested with a simple .rtf and a system .ttf path, and `showrtf` on SDL3 seems to behave the same as SDL2.

**Pull request merged**